youtube_dlc/downloader/hls.py

```diff
@@ -211,6 +222,14 @@ def is_ad_fragment_end(s):
                 elif is_ad_fragment_end(line):
                     ad_frag_next = False
 
+        if external_downloader == 'aria2c':
```

Comment:
Hardcoding `aria2c` everywhere is not the best idea. I'll think of how to implement this better (in a way that makes it is easier to extend the functionality later) and let you know